src/compiler/scala/tools/nsc/typechecker/Typers.scala

```diff
@@ -2942,7 +2942,7 @@ trait Typers extends Adaptations with Tags with TypersTracking with PatternTyper
               else pt // allow type slack (pos/6221)
           }
 
-          ptNorm baseType FunctionSymbol match {
+          unwrapWrapperTypes(ptNorm baseType FunctionSymbol) match {
```

Comment:
This would also have worked.

```scala
(ptNorm baseType FunctionSymbol).typeArgs match {
  case Nil => 
    val dummyPt = if (pt == ErrorType) ErrorType else NoType
    (List.fill(numVparams)(dummyPt), WildcardType) // dummyPt is in CBN position
  case args :+ res =>
    (args, res) 
}
```